app/uk/gov/hmrc/cgtpropertydisposals/service/TaxEnrolmentService.scala

```diff
@@ -72,4 +80,31 @@ class TaxEnrolmentServiceImpl @Inject()(
       case other => Left(Error(s"Received error response from tax enrolment service with http status: $other"))
     }
 
+  @SuppressWarnings(Array("org.wartremover.warts.NonUnitStatements"))
+  def handleDatabaseResult(
+    maybeTaxEnrolmentRequest: Option[TaxEnrolmentRequest]
+  )(implicit hc: HeaderCarrier): Future[Boolean] =
+    maybeTaxEnrolmentRequest match {
+      case Some(enrolmentRequest) =>
+        for {
+          result <- EitherT.liftF(makeTaxEnrolmentCall(enrolmentRequest)) // attempt enrolment
+          _      <- EitherT.fromEither(handleTaxEnrolmentResponse(result)) // evaluate enrolment result
+          _      <- taxEnrolmentRepository.delete(enrolmentRequest.ggCredId) // delete record if enrolment was successful
+        } yield ()
+        Future.successful(true) // always return true regardless of whether the tax enrolment call/delete succeeded or not
```

Comment:
is there a reason you opted for this approach vs. something like:
```scala
val result = for{
  ...
} yield ()
result.leftMap.(e => log.warn(...)).merge
```
The Boolean in this method doesn't serve any useful purpose here. If you want a Boolean in the method below you could yield `maybeEnrolmentRequest.exists` which would make more sense